l10n_br_cnpj_search/models/res_partner_wizard.py

```diff
@@ -0,0 +1,90 @@
+    state_id = fields.Many2one('res.country.state', string="Stade")
+    city_id = fields.Many2one(
+        string="City",
+        comodel_name="res.city",
+        domain="[('state_id', '=', state_id)]",)
+    country_id = fields.Many2one('res.country', string="Country")
+    vat = fields.Char(string="Vat")
+    phone = fields.Char(string="Phone")
+    mobile = fields.Char(string="Mobile")
+    email = fields.Char(string="E-mail")
+    website = fields.Char(string="Website")
+
+
+
+    def default_get(self, fields):
+        res = super(ResPartnerWizard, self).default_get(fields)
+        active_id = self.env.context.get('default_active_id', False)
+
+        partner = self.env['res.partner'].browse(active_id)
+
+        cnpj_cpf = punctuation_rm(partner.cnpj_cpf)
+        webservice = self.env["l10n_br_cnpj_search.webservice.abstract"]
+        response = get(webservice.get_api_url(cnpj_cpf), headers=webservice.get_headers())
+
+        data = webservice.validate(response)
+        values = webservice.import_data(data)
+        res.update({
+            'legal_name': values.get('legal_name', ''),
+            'name': values.get('name',''),
+            'inscr_est' : values.get('inscr_est', ''),
+            'cnpj_cpf': values.get('cnpj_cpf', ''),
+            'zip' : values.get('zip', ''),
+            'street_name': values.get('street_name',''),
+            'street_number' : values.get('street_number', ''),
+            'street2' : values.get('street2',''),
+            'district' : values.get('district', ''),
+            'state_id' : values.get('state_id', ''),
+            'city_id' : values.get('city_id', ''),
+            'country_id' : values.get('country_id', ''),
+            'vat' : values.get('vat', ''),
+            'phone' : values.get('phone', ''),
+            'mobile' : values.get('mobile', ''),
+            'email' : values.get('email',''),
+            'website' : values.get('website', ''),
+            })
+        return res
+
+    def action_update_partner(self):
+        active_id = self.env.context.get('default_active_id', False)
+
+        if active_id:
+            partner = self.env['res.partner'].browse(active_id)
+            partner.write({
+                'cnpj_cpf' : self.cnpj_cpf,
+                'website': self.website,
+                'legal_name': self.legal_name,
+                'name': self.name,
+                'inscr_est' : self.inscr_est,
+                'zip' : self.zip,
+                'street_name': self.street_name,
+                'street_number' : self.street_number,
+                'street2' : self.street2,
+                'district' : self.district,
+                'state_id' : self.state_id,
+                'city_id': self.city_id,
+                'country_id' : self.country_id,
+                'vat' : self.vat,
+                'phone' : self.phone,
+                'mobile' : self.mobile,
+                'email' : self.email,
+                'website' : self.website,
```

Comment:
Conforme solicitado, se o campo estiver vazio não é para atualizá-lo no parceiro.

Reply:
Para fazer essa logica de atualizar somente dados que não esteja em branco, crie um dicionário chamado not_empty_values(valores não vazios) em seguida o código se não tiver valores não vazios a ser atualizado ele chama o método write passando(non_empty_values) como argumento.
Assim ficou o código:
non_empty_values = {key: value for key, value in values_to_update.items() if value}
        if non_empty_values:
            # Update partner only if there are non-empty values
            partner_id.write(non_empty_values)